runtime/parser/parser.go

```diff
@@ -421,31 +424,36 @@ func (p *parser) parseTrivia(options triviaOptions) (containsNewline bool, docSt
 			p.next()
 
 		case lexer.TokenBlockCommentStart:
-			comment := p.parseCommentContent()
-			if options.parseDocStrings {
-				inLineDocString = false
+			commentStartOffset := p.current.StartPos.Offset
+			endToken, ok := p.parseBlockComment()
+
+			if ok && options.parseDocStrings {
+				commentEndOffset := endToken.EndPos.Offset
+
+				contentWithPrefix := p.tokens.Input()[commentStartOffset : commentEndOffset-1]
```

Comment:
No need to rebuild the block comment content with a string builder, we can get the source from the token stream